### ISSUE TITLE
verticalMultilineAtDefnSite breaks if the last parameter on a class has a comment

### ISSUE THREAD

**Issue by pjrt** [scalameta/scalafmt#966]
This template is a guideline, not a strict requirement.
- **Version**: 0.5
- **Integration**: IntelliJ
- **Configuration**:
```
verticalMultilineAtDefinitionSite = true
```

## Steps

I ran scalafmt like this:

```bash
scalafmt
```

## Problem

Original code:

```scala
class Y(
    x: Int,
    y: Int,
    z: Int // Some Comment
)
```

Scalafmt formats code like this:
```scala
class Y(
    x: Int,
    y: Int,
    z: Int // Some Comment)
```
Notice the `)` in the comment. This is no longer valid Scala. 

## Expectation

I would like the formatted output to look like this:
```scala
class Y(
    x: Int,
    y: Int,
    z: Int) // Some Comment
```

## Workaround

Putting the `)` manually before the comment fixes this.

**Comment by pjrt** [comment]
@olafurpg Not sure how to go about this since this requires a rewrite (the comment and the `)` need to be swapped and inlined). Do we do something like this currently? Could you point me to the right place if so?

**Comment by olafurpg** [comment]
Correct, this would be a rewrite. You can look at the other rewrites for examples, it seems this one will need to find those comments that ought to be swapped and then remove the comment token and add the comment syntax to the right of the closing `)` token. The patch api is a bit outdated at the moment, I'm planning to switch soon to the scalafix patch api which I've been polishing for the past months 😎

**Comment by pjrt** [comment]
I see. Should I wait until that's complete then?

**Comment by olafurpg** [comment]
Nah, I will port the rewrites to the new system when I find time to do it. Can't promise when that will happen though

**Comment by pjrt** [comment]
@olafurpg Do we have a system for "required" rewrites? Currently, all rewrites are optional (the user ads them) but this rewrite would need to be mandatory (if `verticalMultilineAtDefinitionSite` is defined). 

If we don't, then I guess I will need to make one.

**Comment by olafurpg** [comment]
Fixed in #1323
